Facebook Ads Cloud Extract updates (#653)
rudderlabs/rudderstack-docs: docs/cloud-extract-sources/facebook-ads.mdx

title: "Facebook Ads"
description: >-
  Step-by-step guide on ingesting data from Facebook Ads into RudderStack.
---

# Facebook Ads

[Facebook Ads](https://www.facebook.com/business/ads) is an online advertising platform that lets you create and run cross-device marketing campaigns and track their performance with easy to read reports.

This document guides you in setting up Facebook Ads as a source in RudderStack. Once configured, RudderStack automatically ingests your specified Facebook Ads data and routes it to your data warehouse.

<div class="warningBlock">

All the Cloud Extract sources support sending data only to a <Link to="/data-warehouse-integrations/">data warehouse destination</Link>.
</div>

## Getting started

To set up Facebook Ads as a source in RudderStack, follow these steps:

1. Log into your [RudderStack dashboard](https://app.rudderstack.com/).
2. Go to **Sources** > **New source** > **Cloud Extract** and select **Facebook Ads** from the list.
3. Assign a name to your source and click on **Continue**.

### Connection settings

1. Click **Connect with Facebook Ads** and grant RudderStack the necessary permissions to access your Facebook Ads data.

> <img src="../assets/cloud-extract-sources/fb-ads-1.png" alt="Facebook Ads connection settings" />

Reviewer: Should we remove the existing acc from this image?

Author: i think that's fine, shouldn't be an issue. I'll move the information box before the image to specify the account and related details automatically appear under "Choose an account".

<div class="infoBlock">

Your Facebook Ads account and the related details will then automatically appear under <strong>Choose an account</strong>.
</div>

2. In the **Source Settings**, configure the following settings:

> <img src="../assets/cloud-extract-sources/fb-ads-2.png" alt="Facebook Ads connection settings" />

Reviewer: may be blurring 'Abhay Tester' from the image

Author: Lol yes - sorry missed out on that. I created a fake account for testing purposes.

- **Page**: Enter the name of the Facebook page from where you want to ingest the Ads data.

> - **Historical data**: Specify the timeframe where RudderStack will fetch all the historical data during the first sync.

Reviewer: ```suggestion
- **Historical data**: Specify the timeframe for which RudderStack should fetch the historical data during the first sync.
```

> - **Attribution window**: Specify the timeframe for any subsequent data syncs.

Reviewer 2: Explanation of Attribution window seems incorrect. Can you check from here? We can add reference link as well.

Author: @RanjeetMishra did you happen to share any reference link? Can't seem to find it in the comment.

Author: Is this the URL? https://www.facebook.com/business/help/2198119873776795?id=768381033531365

<div class="warningBlock">

> RudderStack will not sync the data older than the selected timeframes.

Reviewer 2: `older than the selected timeframes` -> `older than selected timeframe for historical data`?

Author: Makes sense - I will make this change.

> RudderStack will not sync the data older than the selected timeframes.

Author: ```suggestion
RudderStack will not sync the data older than the selected timeframe for the historical data.
```

You can further connect this source to your data warehouse by clicking on **Add Destination**, as shown:

<img src="../assets/cloud-extract-sources/add-destination.png" alt="Adding a destination" />

<div class="successBlock">

Use the <strong>Use Existing Destination</strong> option if you have an already-configured data warehouse destination in RudderStack. To configure a data warehouse destination from scratch, select the <strong>Create New Destination</strong> button.
</div>

## FAQ

### Is it possible to have multiple Cloud Extract sources writing to the same schema?

Yes, it is.

RudderStack associates a table prefix for every Cloud Extract source writing to a warehouse schema. This way, multiple Cloud Extract sources can write to the same schema with different table prefixes.

### What Facebook Ads data does RudderStack import?

| Property         | Table name         | Description    |
| :------------ | :------------- | :------------------ |
| **Account Insights** | **`account_insights`** | This table contains insights, aggregated for the whole account. Insights include total impressions, CPP, CPC, Reach, and CPM.                                                 |

> | **Campaigns**        | **`campaigns`**        | This table holds information information about your campaigns. The columns of this table include: `name`, `objective`, `account_id`, and`status`.                             |

Reviewer: ```suggestion
| **Campaigns**        | **`campaigns`**        | This table holds information about your campaigns. The columns of this table include: `name`, `objective`, `account_id`, and`status`.                             |
```
